src/charm.py

```diff
@@ -140,7 +146,7 @@ def __init__(self, *args):
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
         self.framework.observe(self.on.config_changed, self._configure_sdcore_nms)
-        self._webui = Webui(url="")
+        self._nms = NMS(url=f"http://{self._nms_endpoint}")
```

Comment:
I see a problem here. self._nms_endpoint includes the pod ip. If ip address is not available yet, it can return  the concatenation of  None + : + <NMS_URL_POD> which raises error.

Reply:
How is this a change? we had no validation that this endpoint was a non None value before either. 

Plus the concatenation won't result in an error (I believe), one can concatenate f"{None}:1234", it's just that the API won't answer of course.

I'll resolve the other comments so that we keep the discussion in 1 place.

Reply:
Actually other 2 commits goes together with that one. If we have an endpoint like that and do not update because of the removal of  `self._setup_webui_endpoint_url()' in  Line 165, it will be problematic. If we add Line 165 back then it will be fine.